x-pack/plugins/apm/server/routes/service_groups/get_service_group_alerts.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kqlQuery } from '@kbn/observability-plugin/server';
+import {
+  AggregationsCardinalityAggregate,
+  AggregationsFilterAggregate,
+  QueryDslQueryContainer,
+} from '@elastic/elasticsearch/lib/api/types';
+import { ApmPluginRequestHandlerContext } from '../typings';
+import { SavedServiceGroup } from '../../../common/service_groups';
+
+export async function getServiceGroupAlerts({
+  serviceGroups,
+  authorizedAlertsIndices,
+  context,
+}: {
+  serviceGroups: SavedServiceGroup[];
+  authorizedAlertsIndices?: string[];
+  context: ApmPluginRequestHandlerContext;
+}) {
+  if (
+    !authorizedAlertsIndices ||
+    authorizedAlertsIndices.length === 0 ||
+    serviceGroups.length === 0
+  ) {
+    return {};
+  }
+  const serviceGroupsKueryMap: Record<string, QueryDslQueryContainer> =
+    serviceGroups.reduce((acc, sg) => {
+      return {
+        ...acc,
+        [sg.id]: kqlQuery(sg.kuery)[0],
+      };
+    }, {});
+
+  const esClient = (await context.core).elasticsearch.client;
```

Comment:
Doesn't the ruleDataClient expose a `getReader()` function that you can use, with the indices predefined?

Reply:
Is there any prior art here as to how other teams are doing this?

Reply:
the `alertsClient.find` supports arbitrary aggregations, and handles RBAC and active space filter for us.